src/services/redisSessionStorage.ts

```diff
@@ -5,16 +5,20 @@
 import { Either, isLeft, left, right } from "fp-ts/lib/Either";
 import { ReadableReporter } from "italia-ts-commons/lib/reporters";
 import * as redis from "redis";
-import { isNumber } from "util";
+import { isNumber, promisify } from "util";
 import { SessionToken, WalletToken } from "../types/token";
 import { User } from "../types/user";
 import { log } from "../utils/logger";
 import { ISessionStorage } from "./ISessionStorage";
 
-const sessionKeyPrefix = "SESSION-";
-const walletKeyPrefix = "WALLET-";
+const sessionKeyPrefix = `SESSION-`;
+const walletKeyPrefix = `WALLET-`;
+const userKeyPrefix = `USER-`;
 
 export default class RedisSessionStorage implements ISessionStorage {
+  private sessionNotFoundMessage = "Session not found";
+  private keysAsync = promisify(this.redisClient.keys).bind(this.redisClient);
```

Comment:
Create an `AsyncRedisClient` wrapper class instead, it will be easier to test, something like:

```ts
class AsyncRedisClient {
  constructor(private readonly redisClient: redis.RedisClient) {};
  public getKeys = promisify...
  public setKey = promisify...
}
```

Reply:
@cloudify this change require a complete refactoring of `RedisSessionStorage` and the existing tests. Is required in this story?

Reply:
then I would prefer to avoid changing the behavior of the client, it's too much for this PR, perhaps you can promisify it in a different PR

Reply:
with all these changes is hard to understand what is a refactoring of the previous logic and what is the new logic

Reply:
In this PR should i remove the `keysAsync`, `setAsync` and `getAsync` functions and restore standard calls to `redisClient`?

Reply:
yes we can do a followup PR to change the behavior of the redis client

Reply:
fixed in e55595b